js/views/modals/Settings/Settings.js

```diff
@@ -55,20 +64,90 @@ export default class extends BaseModal {
     }
   }
 
-  cancelClick() {
-    this.cancel();
+  saveClick() {
+    this.save();
   }
 
-  cancel() {
-    this.currentTabView.cancel();
+  save() {
+    this.$save.addClass('loading');
+    $('#statusBarContainer').text('');
+    this.saving = true;
+    this.$saveStatus.text('');
+
+    // Tab views should implement save to return a promise. On errors,
+    // if it's a server error, please return the args of the fail handler
+    // with the promise rejection. Please send a progress event when
+    // client validation succeeds (deferred.notify()).
+    this.currentTabView.save()
+      .always(() => {
+        this.saving = false;
+        this.$save.removeClass('loading');
+      })
+      .fail((...args) => {
+        const $firstErr = this.currentTabView.$('.errorList:first');
+        const isXhr = args[0].abort; // xhr's implement the abort method
+
+        $('#statusBarContainer').text(app.polyglot.t('settings.statusSaveFailed'));
+
+        // sroll to first error
+        if ($firstErr.length) $firstErr[0].scrollIntoViewIfNeeded();
+
+        // on server errors, we'll display a message
+        if (isXhr) {
+          const jqXhr = args[0];
+
+          new SimpleMessage({
+            title: app.polyglot.t('settings.errors.saveError'),
+            message: jqXhr && jqXhr.responseJSON && jqXhr.responseJSON.reason || '',
+          })
+          .render()
+          .open();
+        }
+      })
+      .done(() => {
+        if (this.confirmNavAwayDialog && this.confirmNavAwayDialog.isOpen()) {
+          this.$saveStatus.text(app.polyglot.t('settings.statusSafeToClose'));
+        }
+
+        $('#statusBarContainer').text(app.polyglot.t('settings.statusSaveComplete'));
```

Comment:
I purposely left it like this because this is a very temporary status bar. When I bring in the v1 status bar, we won't even be targeting it this way with a selector.